### PR TITLE
Committing Changes and Adding KeyboardDrive 1-28-22 Jack Tommaney

### PR DESCRIPTION
**Pull request by JackTWing**
Adds KeyboardDrive (yay!!!!!!!!!)

### CODE REVIEW

File: src/main/java/frc/robot/OI.java [hunk 1]
```diff
@@ -8,9 +8,13 @@
 package frc.robot;
 
 import frc.robot.commands.Drive.ArcadeDrive;
+import frc.robot.commands.Drive.KeyboardDrive;
 import frc.robot.commands.Drive.TankDrive;
+import com.fasterxml.jackson.databind.deser.std.StdScalarDeserializer;
 import frc.robot.input.AttackThree;
 import frc.robot.input.ButtonBox;
+import frc.robot.input.KeyboardBox;
+import frc.robot.input.KeyboardBox;
```
Comment (zzberries) [inline src/main/java/frc/robot/OI.java:17]: Please remove double import

File: src/main/java/frc/robot/OI.java [hunk 2]
```diff
@@ -59,6 +67,14 @@ public OI() {
     ////////////////////////////////////////////////////
 
 
+    ////////////////////////////////////////////////////
+    // Now Mapping Commands to Keyboard Box
+    ////////////////////////////////////////////////////
+
+    keyboardBox.getKForward().whileHeld(new KeyboardDrive(0.75, 0));
```
Comment (zzberries) [inline src/main/java/frc/robot/OI.java:74]: Perhaps we can give the constants names (defined in KeyboardDrive) and pass in the named constants

File: src/main/java/frc/robot/commands/Drive/KeyboardDrive.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands.Drive;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.OI;
+import frc.robot.input.KeyboardBox;
+//import frc.robot.TestingDashboard;
```
Comment (zzberries) [inline src/main/java/frc/robot/commands/Drive/KeyboardDrive.java:10]: Please remove the commented-out import

File: src/main/java/frc/robot/commands/Drive/KeyboardDrive.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import frc.robot.input.XboxController;
+import frc.robot.input.XboxController.XboxAxis;
+import frc.robot.subsystems.Drive;
+import frc.robot.testingdashboard.TestingDashboard;
+
+
+public class KeyboardDrive extends CommandBase {
+  /** Creates a new KeyboardDrive. */
+  private final Drive m_drive;
+  private static OI oi;
```
Comment (zzberries) [inline src/main/java/frc/robot/commands/Drive/KeyboardDrive.java:20]: Suggestion: change oi to m_oi (and a similar thing with speed & rotation)

File: src/main/java/frc/robot/input/KeyboardBox.java [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+ * can be accessed through the functions written here.
+ */
+public class KeyboardBox extends Joystick {
+	private JoystickButton WideFocus;
+	private JoystickButton FineFocus;
+	private JoystickButton Fire;
+	private JoystickButton IntakeIn;
+	private JoystickButton IntakeOut;
+	private JoystickButton Hopper;
+	//private JoystickButton CPDeploy;
```
Comment (zzberries) [inline src/main/java/frc/robot/input/KeyboardBox.java:19]: Please remove commented-out code

File: src/main/java/frc/robot/input/KeyboardBox.java [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+	public JoystickButton getHopper() {
+		return this.Hopper;
+	}
+
+	/**
+	 * Gets the Color Panel Deploy button from the ButtonBox
+	 * 
+	 * @return Color Panel Deploy button
+	 */
+//	public JoystickButton getCPDeploy() {
```
Comment (zzberries) [inline src/main/java/frc/robot/input/KeyboardBox.java:115]: Please remove commented-out methods

File: src/main/java/frc/robot/commands/Drive/KeyboardDrive.java (hunk 3 above)
Comment (JackTWing) [reply src/main/java/frc/robot/commands/Drive/KeyboardDrive.java:10]: I think we need it though?

Comment (snowpuppy) [reply src/main/java/frc/robot/commands/Drive/KeyboardDrive.java:10]: We don't. If it's commented out, it's not required for the operation of the software. :)

**Comment by snowpuppy** [comment]
This pull request has been resolved. The commits were merged into the main repository.

**Pull request merged**